python/pyarrow/table.pxi

```diff
@@ -209,10 +209,14 @@ cdef class ChunkedArray(_PandasConvertible):
         -------
         value : Scalar (index) or ChunkedArray (slice)
         """
+        cdef shared_ptr[CScalar] c_scalar
+
         if isinstance(key, slice):
             return _normalize_slice(self, key)
 
-        return self.getitem(_normalize_index(key, self.chunked_array.length()))
+        c_scalar = GetResultValue(self.chunked_array.GetScalar(
+            _normalize_index(key, self.chunked_array.length())))
+        return Scalar.wrap(<shared_ptr[CScalar]> c_scalar)
```

Comment:
I'm curious, why do you need the cast to `shared_ptr[CScalar]`?

Reply:
No need to cast. I refactored `__getitem__()` and `getitem()` methods to follow [the convention of other container classes, eg. `Array`](https://github.com/apache/arrow/blob/master/python/pyarrow/array.pxi#L1139-L1159).